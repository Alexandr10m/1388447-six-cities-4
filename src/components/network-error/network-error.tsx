import * as React from "react";
import {getTextError} from "../../reducer/data/selectors";
import {connect} from "react-redux";


interface Props {
  textError: string;
}

const NetworkError: React.FunctionComponent<Props> = (props: Props) => {
  const {textError} = props;
  return (
    <div className="page page--gray page--main">
      <main className="page__main page__main--index page__main--index-empty">
        <div className="cities">
          <div className="cities__places-container cities__places-container--empty container">
            <section className="cities__no-places">
              <div className="cities__status-wrapper tabs__content">
                <b className="cities__status">{textError}</b>
              </div>
            </section>
            <div className="cities__right-section"></div>
          </div>
        </div>
      </main>
    </div>
  );
};

const mapStateToProps = (state) => ({
  textError: getTextError(state),
});


export {NetworkError};
export default connect(mapStateToProps)(NetworkError);

