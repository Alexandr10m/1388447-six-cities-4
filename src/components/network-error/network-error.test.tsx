import React from "react";
import renderer from "react-test-renderer";
import {NetworkError} from "./network-error.js";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NameSpace from "../../reducer/name-space.js";


const mockStore = configureStore([]);

describe(`Snapshot of NetworkError`, () => {
  it(`NetworkError should render correct`, () => {
    const store = mockStore({
      [NameSpace.STATE]: {},
      [NameSpace.DATA]: {
        textError: `some error`,
      },
      [NameSpace.USER]: {},
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <NetworkError
              textError={`some error`}
            />
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
