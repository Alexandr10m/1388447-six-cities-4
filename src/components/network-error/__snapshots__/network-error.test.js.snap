// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of NetworkError NetworkError should render correct 1`] = `
<div
  className="page page--gray page--main"
>
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <div
      className="cities"
    >
      <div
        className="cities__places-container cities__places-container--empty container"
      >
        <section
          className="cities__no-places"
        >
          <div
            className="cities__status-wrapper tabs__content"
          >
            <b
              className="cities__status"
            >
              some error
            </b>
          </div>
        </section>
        <div
          className="cities__right-section"
        />
      </div>
    </div>
  </main>
</div>
`;
